### ISSUE TITLE
[BUG] - Job with LimitRunsTo(1) runs twice

### ISSUE THREAD

**Issue by Kurt212** [go-co-op/gocron#115]
### Describe the bug

If you add a job to **a running scheduler** and set its **run limits to 1**, then the job is executed **one extra time**. It runs 2 times instead of 1 time.

### To Reproduce

1. StartAsync a scheduler 
2. Add a new job to the scheduler
3. Set the new job's run limit to 1

```go
s := gocron.NewScheduler(time.UTC)

s.StartAsync()

var job *gocron.Job
job, _ = s.Every(1).Second().Do(func() {
  fmt.Println(job.LastRun())
  fmt.Println(job.RunCount())
  fmt.Println()
})

job.LimitRunsTo(1)

time.Sleep(time.Second * 10)
```

The program output:
```
2021-02-04 07:53:52.5575732 +0000 UTC
1

2021-02-04 07:53:53.5576467 +0000 UTC
2
```

### Version

Current master branch. 7601804f2dea4e0e55d415abfb211f8ec68edf14

### Expected behavior

The job is executed exectly once.

### Additional context

Well, I did not investigate deeply, but I guess that is just a mistake in code, because other limits (more than 1) work correctly.

Correct code:
```go
s := gocron.NewScheduler(time.UTC)

s.StartAsync()

var job *gocron.Job
job, _ = s.Every(1).Second().Do(func() {
  fmt.Println(job.LastRun())
  fmt.Println(job.RunCount())
  fmt.Println()
})

job.LimitRunsTo(2) // now set to 2

time.Sleep(time.Second * 10)
```

Output as expected with 2 runs:
```
2021-02-04 08:03:13.5170117 +0000 UTC
1

2021-02-04 08:03:14.5185231 +0000 UTC
2
```
But I also see a different problem with `LimitRunsTo`.

I guess there's a problem with concurrency.

I set job's run limits after the job is added to scheduler. That means that the job may be scheduled an unlimited number of times (theoratically) before the `LimitRunsTo` is executed.

I suggest adding `LimitRunsTo` analog to the new job creating pipeline. Like this:
```go
s.Every(1).Second().LimitRuns(2).Do(func() {})
```

**Comment by arjunmahishi** [comment]
Hey @Kurt212,
As I can see in the above snippets, you are starting the scheduler before adding the job. This works as expected when you first created the job and then start the scheduler.

```go
job, _ := s.Every(1).Second().Do(func() {
	fmt.Println(job.LastRun())
	fmt.Println(job.RunCount())
})
job.LimitRunsTo(1)
s.StartAsync()
```

**Comment by JohnRoesler** [comment]
Hey @Kurt212 

Thanks for bringing this up. As @arjunmahishi points out, it is _working as expected_, but it's super clear why it works that way. 

The reason, is that jobs are started immediately by default, so when you add a job to a running scheduler it will run right away. After the job runs it is immediately rescheduled and because there is no limit, it is scheduled. Then the limit is imposed, and once the 2nd execution occurs, it will check the limit when rescheduling and not allow further runs.

I think the best way for this case - is as you suggest to add to the analog. I'll get that added

**Comment by JohnRoesler** [comment]
This is released in [v0.6.0](https://github.com/go-co-op/gocron/releases/tag/v0.6.0)!
